tools/codegen/pkg/utils/aggregate.go

```diff
@@ -0,0 +1,59 @@
+package utils
+
+import (
+	"bytes"
+	"errors"
+	"strings"
+
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// Aggregate is an error wrapper that prints out aggregate and wrapped
+// errors as individual errors in individual lines.
+type Aggregate struct {
+	inner error
+}
+
+// NewAggregate creates a new Aggregate error.
+func NewAggregate(err error) Aggregate {
+	return Aggregate{
+		inner: err,
+	}
+}
+
+// Error prints the error out as a string.
+// It unwraps wrapped errors and iterates over aggregate errors
+// to print errors on invidividual lines.
+func (a Aggregate) Error() string {
+	buf := bytes.NewBuffer([]byte("\n"))
+
+	handleError(buf, a.inner, 0)
+
+	return buf.String()
+}
+
+// handleError checks if the error given is an aggregate and iterates
+// over its sub-errors if it is, else it checks if it is a wrapped
+// error, in which case it prints out the context and unwraps the error.
+func handleError(buf *bytes.Buffer, err error, indent int) {
```

Comment:
nit: Would it make sense to limit recursion depth here? Check the indent value and maybe panic if its > 100 or so. But, given how and where this code is used, I'm not too concerned about this.

Reply:
That's a good idea, or just, print out the error when it's at 100 levels